src/core/CachedPackageLocator.js

```diff
@@ -0,0 +1,61 @@
+import path from 'path'
+import os from 'os'
+import fs from 'fs'
+
+export default class CachedPackageLocator {
```

Comment:
i wonder if this would be better extracted to its own published package with its own tests - is that something you'd be willing to publish and maintain?

Reply:
Hmm sure, I'll try to find some some for this either today or this week.